cli/internal/hashing/package_deps_hash.go

```diff
@@ -416,3 +432,17 @@ func gitStatus(rootPath turbopath.AbsolutePath, patterns []string) (map[turbopat
 
 	return output, nil
 }
+
+// appends an item into a copy of an array to prevent mutating the original.
+// uses generics to take an array of any type and append an item of the same type.
+func copyAndAppend[K comparable](input []K, item K) []K {
```

Comment:
`copyAndAppend` is a _bit_ too clever for Go.

Couple of things:
- I'm sorry, I forgot about `copy`, I'm usually doing a transformation of the slice contents along the way or pulling them out of a map. This case is different which leaves `copy` available.
- Even if there weren't `copy` Go _loves_ verbosity so I generally would choose a `copy` then `append` pattern.

```go
a := []int{1, 2, 3}
b := make([]int, len(a))
copy(b, a)
b = append(b, 6)
b[1] = 5
fmt.Printf("a: %v b: %v", a, b)
```

Reply:
inlined, but chose to always copy the input, rather than conditionally. hope that's ok